extension/js/common/platform/store/in-memory-store.ts

```diff
@@ -2,19 +2,42 @@
 
 import { AbstractStore } from './abstract-store.js';
 import { Time } from '../../browser/time.js';
-import { BrowserMsg } from '../../browser/browser-msg.js';
+import { BgNotReadyErr, BrowserMsg } from '../../browser/browser-msg.js';
+import { Ui } from '../../browser/ui.js';
 
 /**
  * Temporary In-Memory store for sensitive values, expiring after clientConfiguration.in_memory_pass_phrase_session_length (or default 4 hours)
  * see background_page.ts for the other end, also ExpirationCache class
  */
 export class InMemoryStore extends AbstractStore {
   public static async set(acctEmail: string, key: string, value?: string, expiration?: number) {
-    return await BrowserMsg.send.bg.await.inMemoryStoreSet({ acctEmail, key, value, expiration });
+    for (let i = 0; i < 10; i++) {
+      // backend may not be immediately ready to respond - retry
+      try {
+        return await BrowserMsg.send.bg.await.inMemoryStoreSet({ acctEmail, key, value, expiration });
+      } catch (e) {
+        if (!(e instanceof BgNotReadyErr) || i === 9) {
+          throw e;
+        }
+        await Ui.delay(300);
+      }
+    }
+    throw new BgNotReadyErr('this should never happen');
   }
 
   public static async get(acctEmail: string, key: string): Promise<string | undefined> {
-    return (await BrowserMsg.send.bg.await.inMemoryStoreGet({ acctEmail, key })) ?? undefined;
+    for (let i = 0; i < 10; i++) {
+      // backend may not be immediately ready to respond - retry
+      try {
+        return (await BrowserMsg.send.bg.await.inMemoryStoreGet({ acctEmail, key })) ?? undefined;
+      } catch (e) {
+        if (!(e instanceof BgNotReadyErr) || i === 9) {
+          throw e;
+        }
+        await Ui.delay(300);
+      }
+    }
+    throw new BgNotReadyErr('this should never happen');
```

Comment:
works well 👍 I've just noticed that `BgNotReadyErr` happens for `expirationCacheDeleteExpired`, so probably it needs similar retry mechanism too:
```
Uncaught (in promise)     at BrowserMsg.sendRaw (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/content_scripts/webmail_bundle.js:10298:26)
    at BrowserMsg.sendAwait (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/content_scripts/webmail_bundle.js:10286:37)
    at Object.expirationCacheDeleteExpired (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/content_scripts/webmail_bundle.js:10445:62)
    at ExpirationCache.deleteExpired (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/content_scripts/webmail_bundle.js:3474:52)
    at MessageRenderer.deleteExpired (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/content_scripts/webmail_bundle.js:12550:41)
    at GmailElementReplacer.everything (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/content_scripts/webmail_bundle.js:14809:34)
    at Object.handler (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/content_scripts/webmail_bundle.js:14735:55)
    at Object.start (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/content_scripts/webmail_bundle.js:16354:46)

url: https://mail.google.com/mail/u/0/#inbox


Error: BgNotReadyErr: BrowserMsg.sendAwait(expirationCacheDeleteExpired) failed with lastError: Could not establish connection. Receiving end does not exist.
    at processRawMsgResponse (chrome-extension://pjohdcpiapjhagmakkjpgjmodmgfmkkd/js/content_scripts/webmail_bundle.js:10334:33)
    at <async> sendAwait (/extension/js/common/browser/browser-msg.ts:446:4)
    at <async> deleteExpired (/extension/js/common/core/expiration-cache.ts:59:25)
```

Reply:
Sounds good